server/main.py

```diff
@@ -44,13 +46,34 @@ async def send(self, data: EventResponse) -> None:
         """
         await self._websocket.send_json(data.dict())
 
-    async def receive(self) -> EventRequest:
+    async def receive(self) -> EventRequest | None:
         """Receives JSON data over the WebSocket connection.
 
         Returns:
-            The data received from the client.
+            The data received from the client or None if an error occured.
         """
-        return EventRequest(**await self._websocket.receive_json())
+        try:
+            return EventRequest(**await self._websocket.receive_json())
+        except TypeError:
+            await self.send(
+                EventResponse(
+                    type=EventType.ERROR,
+                    data=ErrorData(message="Invalid request data."),
+                    status_code=StatusCode.INVALID_REQUEST_DATA,
+                ),
+            )
+            return
+        except (KeyError, ValidationError, JSONDecodeError):
```

Comment:
I think that the `JSONDecodeError` should have a status code of `INVALID_REQUEST_DATA`.

It could be just moved up with `TypeError`:
```py
except (TypeError, JSONDecodeError):
    ...
```